trawler/src/main/java/io/m9sweeper/trawler/framework/docker/DockerImage.java

```diff
@@ -57,6 +57,38 @@ public void setName(String name) {
         this.name = name;
     }
 
+    /**
+     * Builds the full path that can be used in a Docker pull command
+     * @param includeRegistry Whether it should include the registry prefix
+     * @param includeHash Whether it should include the sha. Will not include the sha if it is null or starts with TMP_ regardless of the value
+     * @return the full path of the docker image
+     */
+    public String buildFullPath(boolean includeRegistry, boolean includeHash) {
+        StringBuilder builder = new StringBuilder(32);
+        if (includeRegistry) {
+            builder.append(this.registry.getHostname())
+                    .append('/');
+        }
+
+        builder.append(this.name)
+                .append(":")
+                .append(this.tag);
```

Comment:
The tag is ignored when a SHA is present, so I left it as always being part of the string.